runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -1281,15 +1295,27 @@ bool
 ROMClassBuilder::compareROMClassForEquality(U_8 *romClass,bool romClassIsShared,
 		ROMClassWriter *romClassWriter, SRPOffsetTable *srpOffsetTable, SRPKeyProducer *srpKeyProducer,
 		ClassFileOracle *classFileOracle, U_32 modifiers, U_32 extraModifiers, U_32 optionalFlags,
-		ROMClassCreationContext * context)
+		ROMClassCreationContext * context, U_32 romSize, bool isLambda)
 {
-	ComparingCursor compareCursor(_javaVM, srpOffsetTable, srpKeyProducer, classFileOracle, romClass, romClassIsShared, context);
+	if(isLambda && sizeof(U_64) < abs((int)(romSize - ((J9ROMClass *)romClass)->romSize))){
+		/* if the class is a lambda class, compare the romSizes first to avoid checking everything */
+		return false;
+	}
+
+	if(isLambda){
+		/* if it is a lambda class we already compared the romSizes so we assign the romSize of the compared class to match them */
+		romSize = ((J9ROMClass *)romClass)->romSize;
```

Comment:
If shouldCheckForEquality() returns false for ROM_SIZE, there is no need to set romSize here.